### ISSUE TITLE
Incorrect warning during check for manifest specification in Component due to single quotes

### ISSUE THREAD

**Issue by qualiture** [UI5/linter#306]
## Expected Behavior

When specifying in the Component that it uses the descriptor via the `manifest.json` file, it should not matter if you use single quotes or double quotes, i.e. `manifest: "json"` or `manifest: 'json'` should both be correct.

## Current Behavior

When specifying the manifest with double quotes, i.e.

```typescript
    public static readonly metadata = {
        manifest: "json"
    };
```
the linter throws no warning.

However, when using single quotes, i.e.

```typescript
    public static readonly metadata = {
        manifest: 'json'
    };
```
it throws the following warning:

`warning Component does not specify that it uses the descriptor via the manifest.json file. Details: A manifest.json has been found in the same directory as the component. Although it will be used at runtime automatically, this should still be expressed in the metadata of the component class (https://ui5.sap.com/#/topic/0187ea5e2eff4166b0453b9dcc8fc64f).`

## Steps to Reproduce the Issue

In the `Component.{js|ts}` file, use single quotes to specify the manifest json descriptor

## Context

- UI5 linter version: **`0.3.5`**
- Node.js Version: **`20.17.0`**
- npm Version: `10.8.2`
- OS/Platform: `SAP BAS vm`
- Other information regarding your environment *(optional)*: Tested with a Typescript application

## Log Output / Stack Trace

N/A

**Comment by codeworrior** [comment]
From your code samples, I assume you're using TypeScript for your project, together with the ui5-tooling-transpile-task?

If so, my guess would rather be that the UI5 linter does not (yet) support the TS structure of the Component.ts (it differs from the structure in the Component.js - `metadata` there is a static property, not a property of the object given to the extend` call).

On what sources do you execute the UI5 linter? On the original *.ts sources or on the transpiled *.js files?

**Comment by qualiture** [comment]
Hi @codeworrior, 

It is indeed a Typescript project with `ui5-tooling-transpile-task`.

I tried with all combinations of 
`public metadata = {`, 
`public static metadata = {` and 
`public static readonly metadata = {`, 

and in all cases it stumbled over the single quotes around 'json' in `manifest: 'json'`

While any of the above combinations with double quotes work perfectly fine, and no warning is given...

As our team enforces the use of single quotes in Javascript / Typescript applications via Prettier, as a workaround I have for now used the prettier-ignore comment:

```typescript
    public static readonly metadata = {
        // prettier-ignore
        manifest: "json"
    };
```

**Comment by RandomByte** [comment]
Fix has been released as part of v0.4.0. Please let us know in case you still experience this issue, thanks!
